src/IRMatch.h

```diff
@@ -200,24 +200,24 @@ struct SpecificExpr {
 
     constexpr static uint32_t binds = 0;
 
-    const BaseExprNode &expr;
+    Expr expr;
```

Comment:
Given the subtlety of the failure, it's probably worth adding something similar to the PR description here for future maintainers

Reply:
Capturing an Expr seems totally reasonable. Really the original should have had a good justification for why capturing a const BaseExprNode & was safe.